new-log-viewer/src/typings/states.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Various states of the UI. UI components may be enabled or disabled based on the state.
+ */
+enum UI_STATE {
+
+    /**
+     * When there is no file opened in the viewer.
+     */
+    UNOPENED,
+
+    /**
+     * When a file load request is pending response.
+     */
+    FILE_LOADING,
+
+    /**
+     * When a fast request is pending response. In this state, UI elements are not visually
+     * disabled but instead ignore pointer events. Rapidly disabling/enabling UI elements is
+     * jarring to the user.
+     */
+    FAST_LOADING,
+
+    /**
+     * When a slow request is pending response.
+     */
+    SLOW_LOADING,
+
+    /**
+     * When the file is loaded.
+     */
+    READY,
+}
+
+/**
+ * Stateful elements in the UI.
+ */
+enum UI_ELEMENT {
+    PROGRESS_BAR,
+    NAVIGATION_BAR,
+    OPEN_FILE_BUTTON,
+    LOG_LEVEL_FILTER,
+    EXPORT_LOGS_BUTTON,
+    LOG_EVENT_NUM_DISPLAY,
+    DRAG_AND_DROP,
```

Comment:
if alphabetizing those makes sense
```suggestion
    DRAG_AND_DROP,
    EXPORT_LOGS_BUTTON,
    LOG_EVENT_NUM_DISPLAY,
    LOG_LEVEL_FILTER,
    NAVIGATION_BAR,
    OPEN_FILE_BUTTON,
    PROGRESS_BAR,
```